tools/monorepo-utils/src/ci-jobs/lib/job-processing.ts

```diff
@@ -302,13 +302,13 @@ async function createJobsForProject(
 			dependencyCascade
 		);
 
-		if (
-			dependencyChanges === false &&
-			Object.values( dependencyJobs ).some(
-				( array ) => array.length > 0
-			)
-		) {
-			dependencyChanges = true;
+		if ( dependencyChanges === false ) {
+			// First line of detection: implicit changes list points to the dependency.
+			dependencyChanges = ( changes[ dependency.name ] || [] ).length > 0;
+			if ( dependencyChanges === false ) {
+				// Second line of detection: the dependency spawns testing jobs.
+				dependencyChanges = dependencyJobs.test.length > 0;
```

Comment:
On the second line of detection, I think we should include any job type, including lint.

Reply:
Addressed in 7224d30a4a94723046547d5c9fa6c68dfe054112